app/ts/background/messageSending.ts

```diff
@@ -0,0 +1,55 @@
+import { InterceptedRequestForward, InterceptorMessageToInpage, SubscriptionReplyOrCallBack } from "../utils/interceptor-messages.js"
+import { WebsiteSocket, WebsiteTabConnections } from "../utils/user-interface-types.js"
+import { websiteSocketToString } from "./backgroundUtils.js"
+
+function postMessageToPortIfConnected(port: browser.runtime.Port, message: InterceptorMessageToInpage) {
+	try {
+		port.postMessage(InterceptorMessageToInpage.serialize(message) as Object)
+	} catch (error) {
+		if (error instanceof Error) {
+			if (error.message?.includes('Attempting to use a disconnected port object')) {
+				return
+			}
+			if (error.message?.includes('Could not establish connection. Receiving end does not exist')) {
+				return
+			}
+		}
+		throw error
+	}
+}
+export function postMessageIfStillConnected(websiteTabConnections: WebsiteTabConnections, socket: WebsiteSocket, message: InterceptorMessageToInpage) {
+	const tabConnection = websiteTabConnections.get(socket.tabId)
+	const identifier = websiteSocketToString(socket)
+	if (tabConnection === undefined) return false
+	for (const [socketAsString, connection] of Object.entries(tabConnection.connections)) {
```

Comment:
This is dangerous as the `Object.entries` return type is not sound.  Take the following example:
```ts
type Fruit = { type: 'apple' | 'banana', color: 'red' | 'yellow' }
function eat(fruit: Fruit) {
	const entries = Object.entries(fruit)
	for (const [ key, value ] of entries) {
		// compiletime: typeof key === 'type' | 'color'
		// runtime: typeof key === 'type' | 'color' | 'seed'
	}
}
const apple = { type: 'apple', color: 'red', seeds: true } as const
eat(apple)
```
Generally speaking, you should avoid using `Object.entries` unless you are *absolutely certain* that there is no way (now or in the future) that extra properties can't be added to by someone in the future, because the type checker won't validate it correctly.  Since `tabConnection.connections` is coming from somewhere else, I strongly recommend not using `Object.entries` here and instead iterating over opaque keys and narrowing inside the loop.